packages/phone-number-privacy/signer/src/signing/get-partial-signature.ts

```diff
@@ -79,16 +79,35 @@ export async function handleGetBlindedMessagePartialSig(
       )
       return
     }
+
     const keyProvider = getKeyProvider()
     const privateKey = keyProvider.getPrivateKey()
     const signature = computeBlindedSignature(blindedQueryPhoneNumber, privateKey)
-    if (!(await getRequestExists(request.body))) {
-      await incrementQueryCount(account)
-      await storeRequest(request.body)
+
+    let didIncrementQueryCount: boolean = false
+    let didStoreRequest: boolean = false
+
+    if (await getRequestExists(request.body)) {
+      logger.debug('Signature request already exists in db.')
+      errorMsg = WarningMessage.DUPLICATE_REQUEST_TO_GET_PARTIAL_SIG
     } else {
-      logger.debug('Signature request already exists in db. Leaving quota unchanged.')
+      didStoreRequest = !!(await storeRequest(request.body))
+      if (!didStoreRequest) {
+        errorMsg = WarningMessage.FAILURE_TO_STORE_REQUEST
+      }
+      didIncrementQueryCount = !!(await incrementQueryCount(account))
+      if (!didIncrementQueryCount) {
+        errorMsg = WarningMessage.FAILURE_TO_INCREMENT_QUERY_COUNT
+      }
+    }
+    if (!didStoreRequest) {
+      logger.debug('Did not store request.')
+    }
+    if (!didIncrementQueryCount) {
+      logger.debug('Did not increment query count.')
```

Comment:
Can these be DRY'ed up with the two if statements above which check the same thing?

Reply:
I wrote it this way so that the logging will happen regardless of why the update didn't happen. That is, if the db already contained the request vs. the db insert just failed.

Reply:
My intuition is that there may be a more concise way to structure this but it's really nbd :)